aws-auth-cognito/src/test/java/featureTest/utilities/CognitoMockFactory.kt

```diff
@@ -63,61 +52,35 @@ class CognitoMockFactory(
     private val mockCognitoIPClient: CognitoIdentityProviderClient,
     private val mockCognitoIdClient: CognitoIdentityClient
 ) {
-    private val captures: MutableMap<String, CapturingSlot<*>> = mutableMapOf()
-
     fun mock(mockResponse: MockResponse) {
         val responseObject = mockResponse.response as JsonObject
 
-        return when (mockResponse.apiName) {
+        when (mockResponse.apiName) {
             "forgotPassword" -> {
-                val requestBuilderCaptor = slot<ForgotPasswordRequest>()
-
-                coEvery { mockCognitoIPClient.forgotPassword(capture(requestBuilderCaptor)) } coAnswers {
-                    if (mockResponse.responseType == ResponseType.Failure) {
-                        throw Json.decodeFromString(
-                            CognitoIdentityProviderExceptionSerializer,
-                            responseObject.toString()
-                        )
-                    }
+                coEvery { mockCognitoIPClient.forgotPassword(any()) } coAnswers {
```

Comment:
Dont we have to capture requests too so that we can validate cognito calls are made with expected input params?

Reply:
I hope you found the answer, but we use verify instead of captors to check that.